docker/DockerfileNoetic

```diff
@@ -0,0 +1,77 @@
+# Start with ROS base image
+FROM ros:noetic-ros-base
+
+# Tools useful during development.
+RUN apt update \
+ && apt install -y --no-install-recommends\
+        cppcheck \
+        curl \
+        cmake \
+        lsb-release \
+        gdb \
+        git \
+        python3-dbg \
+        python3-pip \
+        python3-venv \
+        ruby \
+        software-properties-common \
+        sudo \
+        wget \
+        libeigen3-dev \
+        pkg-config \
+        protobuf-compiler \
+ && apt clean
+
+# Set distributions
+ARG DIST=noetic
+ARG GAZ=gazebo11
+
+
+# setup sources.list
+# ROS
+RUN echo "deb http://packages.ros.org/ros/ubuntu focal main" > /etc/apt/sources.list.d/ros1-latest.list
+# Gazebo
+RUN . /etc/os-release \
+    && echo "deb http://packages.osrfoundation.org/gazebo/$ID-stable `lsb_release -sc` main" > /etc/apt/sources.list.d/gazebo-latest.list
+
+
+# setup keys
+# ROS
+RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654
+# Gazebo
+RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys D2486D2DD83DB69272AFE98867170598AF249743
+
+# setup environment
+ENV LANG C.UTF-8
+ENV LC_ALL C.UTF-8
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    qtbase5-dev \
+    libgles2-mesa-dev \
+    ${GAZ} \
+    lib${GAZ}-dev \
+    ros-${DIST}-xacro \
```

Comment:
These are some common tools. I don't know if they're necessary, if Michael is able to get everything working. I just see it in my standard Dockerfile.
Maybe you don't want these if you are looking to keep dependencies to a minimum. They can always be added later.
```suggestion
    python3-catkin-tools \
    python3-rosdep \
    python3-rosinstall \
    python3-rosinstall-generator \
    python3-wstool \
    ros-${DIST}-xacro \
```

Reply:
I think it's good to include them. We are not trying to make this image super lean.